flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java

```diff
@@ -317,4 +320,25 @@ private void waitUntilAllTaskManagerConnected() throws InterruptedException, Tim
                 DEFAULT_TIMEOUT,
                 "TaskManagers are not ready within 30 seconds");
     }
+
+    private void deleteJobManagerTemporaryFiles() {
+        try {
+            final Container.ExecResult result =
+                    jobManager.execInContainer(
+                            "rm",
+                            "-rf",
+                            Paths.get(conf.get(CheckpointingOptions.CHECKPOINTS_DIRECTORY), "*")
+                                    .toString(),
+                            Paths.get(conf.get(HighAvailabilityOptions.HA_STORAGE_PATH), "*")
+                                    .toString());
+            if (result.getExitCode() != 0) {
+                throw new IllegalStateException(
+                        String.format(
+                                "rm command returned non-zero exit code %d. \nSTDOUT: %s\nSTDERR: %s",
+                                result.getExitCode(), result.getStdout(), result.getStderr()));
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to delete temporary files in JobManager");
```

Comment:
Nit: Maybe add the directory names to the exception.

Reply:
We should not swallow the causing and exception and pass it to the RuntimeException